test/unit/test_semver.py

```diff
@@ -134,3 +136,39 @@ def test__resolve_to_specific_version(self):
                 create_range(None, '<=0.0.5'),
                 ['0.0.3', '0.1.4', '0.0.5']),
             '0.0.5')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('=0.4.5a2', '=0.4.5a2'),
+                ['0.4.5a1', '0.4.5a2']),
+            '0.4.5a2')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('=0.7.6', '=0.7.6'),
+                ['0.7.6-b1', '0.7.6']),
+            '0.7.6')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('>=1.0.0', None),
+                ['1.0.0', '1.1.0a1', '1.1.0', '1.2.0a1']),
+            '1.2.0a1')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('>=1.0.0', '<1.2.0'),
+                ['1.0.0', '1.1.0a1', '1.1.0', '1.2.0a1']),
+            '1.1.0')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('>=1.0.0', None),
+                ['1.0.0', '1.1.0a1', '1.1.0', '1.2.0a1', '1.2.0']),
+            '1.2.0')
+
+        self.assertEqual(
+            resolve_to_specific_version(
+                create_range('>=1.0.0', '<1.2.0'),
+                ['1.0.0', '1.1.0a1', '1.1.0', '1.2.0a1', '1.2.0']),
+            '1.1.0')
```

Comment:
@jtcohen6 does the behaviour of the above four test cases make sense to you?

Reply:
Those all look spot on!